File: MotivApp/app/src/main/java/com/motiv/piotr/dao/RateLimitDao.java
package com.motiv.piotr.dao;

import androidx.room.*;
import com.motiv.piotr.RateLimit;

@Dao
public interface RateLimitDao {

    @Insert
    public long saveRateLimit(com.motiv.piotr.RateLimit rateLimit);

    @Delete
    public int deleteRateLimit(com.motiv.piotr.RateLimit rateLimit);

    @Query("SELECT * FROM rateLimit")
    public java.util.List<RateLimit> getRateLimits();

    @Query("SELECT * FROM rateLimit WHERE id = :id")
    public RateLimit getRateLimit(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveRateLimits(java.util.List<com.motiv.piotr.RateLimit> rateLimit);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/RateLimitDao.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.RateLimit
import kotlin.collections.List
@Dao
public interface RateLimitDao {

    @Insert
    public fun saveRateLimit(rateLimit: com.motiv.piotr.RateLimit): Long@Delete
    public fun deleteRateLimit(rateLimit: com.motiv.piotr.RateLimit): Int@Query("SELECT * FROM rateLimit")
    public fun getRateLimits(): List<RateLimit>@Query("SELECT * FROM rateLimit WHERE id = :id")
    public fun getRateLimit(id: String): RateLimit@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveRateLimits(rateLimit: List<com.motiv.piotr.RateLimit>): Unit
}
